test.js

```diff
@@ -157,6 +157,19 @@ test('destroy', (t) => {
   })
 })
 
+test('destroy while opening', (t) => {
+  t.plan(2)
+
+  const dest = file()
+  const stream = new SonicBoom(dest)
+
+  t.ok(stream.write('hello world\n'))
+  stream.destroy()
+  stream.on('finish', () => {
+    t.pass('finish emitted')
```

Comment:
As far as I understand this, the file should not be written. Can you add an assertion to check that?

Reply:
Whether it is written or not is undefined. Though in his specific case I don’t think it is possible not to write without a performance overhead.

I think that’s a separate issue. I should probably just remove the write part from the test.

Reply:
Thanks, that will be better.